File: pinyougou_controller_manager_337/src/main/webapp/js/service/brandService.js
//自定义服务层: 发起请求的
app.service("brandService",function ($http) {
    //查询所有的品牌
    //this代表当前服务层即service
    this.findAll = function () {
        return $http.get("/brand/findAll.do");
    };

    //根据条件查询分页对象
    this.search = function (currentPage,itemsPerPage,searchEntity) {
        return  $http.post("/brand/search.do?pageNum="+currentPage +"&pageSize="+itemsPerPage,searchEntity);

    };

    //查询分页对象
    this.findAllByPage = function (currentPage,itemsPerPage) {
        return $http.get("/brand/findAllByPage.do?pageNum="+currentPage +"&pageSize="+itemsPerPage);
    };

    //新增
    this.save = function (entity) {
        return $http.post("/brand/save.do",entity);
    };

    //修改
    this.update = function (entity) {
        return $http.post("/brand/update.do",entity);
    };

    //根据id查询一个品牌并回显
    this.findById = function (id) {
        return $http.get("/brand/findById.do?id="+id);
    };

    //删除
    this.delete = function (selectedIds) {
        return $http.get("/brand/deletes.do?ids="+selectedIds);
    };

    //品牌下拉列表数据
    this.selectOptionList = function(){
        return $http.get("../brand/selectOptionList.do");//url 入参：无(因为查询所有)  返回值：List<Map>
    }

});

File: pinyougou_interface_337/src/main/java/com/itheima/core/service/BrandService.java
package com.itheima.core.service;

import com.itheima.core.pojo.good.Brand;
import entity.PageResult;

import java.util.List;
import java.util.Map;

/**
 * @描述:
 * @Auther: yanlong
 * @Date: 2019/3/1 19:07:32
 * @Version: 1.0
 */
public interface BrandService {

    /**
     *@描述: 查询所有品牌
     *@Param: []
     *@Return: java.util.List<com.itheima.core.pojo.good.Brand>
    */
    List<Brand> findAll();

    /**
     *@描述: 根据分页查询所有品牌
     *@Param: [pageNum, pageSize]
     *@Return: entity.PageResult
    */
    PageResult findAllByPage(Integer pageNum, Integer pageSize);

    /**
     *@描述: 保存品牌
     *@Param: [brand]
     *@Return: void
    */
    void save(Brand brand);

    /**
     *@描述: 根据id查询一个品牌并回显
     *@Param: [id]
     *@Return: com.itheima.core.pojo.good.Brand
    */
    Brand findById(Long id);

    /**
     *@描述: 修改品牌
     *@Param: [brand]
     *@Return: void
    */
    void update(Brand brand);

    /**
     *@描述: 删除品牌 或 批量删除
     *@Param: [ids]
     *@Return: void
    */
    void deletes(Long[] ids);

    /**
     *@描述: 根据条件查询分页对象
     *@Param: [pageNum, pageSize, brand]
     *@Return: entity.PageResult
    */
    PageResult search(Integer pageNum, Integer pageSize, Brand brand);

    List<Map> selectOptionList();
}
